owa/app/js/components/manage_patient_identifier_sources/edit_patient_identifier_sources/remoteIdentifierSource.jsx

```diff
@@ -0,0 +1,103 @@
+import React, {Component} from 'react';
+import {Table, Button, Input} from 'reactstrap';
+import apiCall from '../../../utilities/apiHelper';
+import './editPatientIdentifierSources.css'
+
+export default class EditRemoteIdentifierSource extends Component {
+
+  constructor(props){
+    super(props);
+    this.state = {
+      index: props.identifierSource.index,
+      uuid: props.identifierSource.uuid,
+      name: props.identifierSource.name,
+      description: props.identifierSource.description,
+      identifierTypeValidator: props.identifierSource.identifierTypeValidator,
+      url: props.identifierSource.url,
+      user: props.identifierSource.user,
+      password: props.identifierSource.password
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event){
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  handleSubmit(){
+    let data = {
+      name: this.state.name,
+      description: this.state.description,
+      url: this.state.url,
+      user: this.state.user,
+      password: this.state.password
+    };
+
+    apiCall(data, 'post', '/idgen/identifiersource/' + this.state.uuid).then((response) => {
+      if( response["error"] === undefined ) {
```

Comment:
I think sometimes when `error` is not present, it can return `null` or `undefined`. So I think you should try doing something like `if (!response.error)`. This should take care of `null` and `undefined`

Reply:
fixed